app/goal_routes.py

```diff
@@ -0,0 +1,92 @@
+from flask import Blueprint, jsonify, request, make_response, abort
+from app.models.task import Task
+from app.models.goal import Goal
+from app.routes import validate_model_id
+from sqlalchemy import desc, asc
+import datetime
+import requests
+import os
+from app import db
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+@goals_bp.route("", methods=["GET"])
+def retrieve_goals():
+    goals = Goal.query.all()
+    goals_response = []
+
+    for goal in goals:
+        goals_response.append(goal.format_goal())
+    return jsonify(goals_response)
+
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+
+    if not request_body.get("title"):
+        return {"details": "Invalid data"}, 400
+
+    new_goal = Goal(title=request_body["title"])
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal": {"id": new_goal.goal_id, "title": new_goal.title}}, 201
+
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def handle_goal(goal_id):
+    goal = validate_model_id(Goal, goal_id)
+
+    return {"goal": goal.format_goal()}
+
+
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = validate_model_id(Goal, goal_id)
+
+    request_body = request.get_json()
+
+    goal.title = request_body["title"]
+    db.session.commit()
+
+    return make_response({"goal": {"id": goal.goal_id, "title": goal.title}})
```

Comment:
Like my comment on missed `format_task` opportunities, `format_goal` could have been used here.

Reply:
And other places, but I'll not repeat myself.